`snlls`: Fix violation of boundaries due to float-point round-off errors

luisfabib opened the pull request:
Closes #177

stestoll commented on deerlab/utils/utils.py:
```diff
@@ -408,7 +408,15 @@ def Jacobian(fcn, x0, lb, ub):
+    except ValueError as err:
+        if '`x0` violates bound constraints.' in str(err):
+            # Protect against seldom float-point round-off errors
+            x0,lb,ub = [var.round(12) for var in [x0,lb,ub]]
```
This looks like a hack. What are the conditions that trigger this error?

luisfabib replied:
This is casused by a round-off error. During the Jacobian evulation for the uncertianty propagation one of the values is evaluated at `x=0.9999999999999964` but its bounds are `[1.0,inf]` leading to `x<lb` and to the error by `scipy.optimize._numdiff.approx_derivative`

_Originally posted by @luisfabib in https://github.com/JeschkeLab/DeerLab/issues/177#issuecomment-868276540_

stestoll commented on deerlab/utils/utils.py:
```diff
@@ -408,7 +408,15 @@ def Jacobian(fcn, x0, lb, ub):
-    J = opt._numdiff.approx_derivative(fcn,x0,method='2-point',bounds=(lb,ub))
+    try:
+        J = opt._numdiff.approx_derivative(fcn,x0,method='2-point',bounds=(lb,ub))
+    except ValueError as err:
```
Add a test where this if branch gets triggered.

luisfabib commented:
After digging a bit more, the float-point error appears in one of the variables returned by the `scipy.optimize.lsq_linear` function. I removed the changed to `Jacobian` and added an explicit control to the results of the linear LSQ results in `snlls`.
